File: leetcode/0171. Excel Sheet Column Number/Solution.swift
// https://leetcode.com/problems/excel-sheet-column-number/

class Solution {
    func titleToNumber(_ s: String) -> Int {
        let characters = Array(s).map { Character(String($0)) }
        let a_ord = Character("A").asciiValue ?? 0
        var multiplier = 1
        var result = 0
        for character in characters.reversed() {
            let character_ord = character.asciiValue ?? 0
            let difference = Int(character_ord - a_ord)
            result += (difference + 1) * multiplier
            multiplier *= 26
        }

        return result
    }
}

let s = Solution()
assert(s.titleToNumber("A") == 1)
assert(s.titleToNumber("B") == 2)
assert(s.titleToNumber("C") == 3)
assert(s.titleToNumber("Z") == 26)
assert(s.titleToNumber("AA") == 27)
assert(s.titleToNumber("AB") == 28)
assert(s.titleToNumber("ZY") == 701)
assert(s.titleToNumber("ZZ") == 702)

File: leetcode/0171. Excel Sheet Column Number/Solution.py
# https://leetcode.com/problems/excel-sheet-column-number/

class Solution:
    def titleToNumber(self, s: str) -> int:
        multiplier = 1
        result = 0
        for c in reversed(list(s)):
            difference = ord(c) - ord("A")
            result += (difference + 1) * multiplier # A = 1
            multiplier *= 26

        return result

s = Solution()
assert s.titleToNumber("A") == 1
assert s.titleToNumber("B") == 2
assert s.titleToNumber("C") == 3
assert s.titleToNumber("Z") == 26
assert s.titleToNumber("AA") == 27
assert s.titleToNumber("AB") == 28
assert s.titleToNumber("ZY") == 701
assert s.titleToNumber("ZZ") == 702
